app/controllers/application_controller.rb

```diff
@@ -11,16 +11,30 @@ def current_user
   end
   helper_method :current_user
 
+  def current_admin=(admin)
+    session[:admin_id] = admin.id
+  end
+
+  def current_admin
+    @current_admin ||= Admin.find_by(id: session[:admin_id])
+  end
+  helper_method :current_admin
+
   # Returns a boolean representing if the user is logged in
   def logged_in?
     !!current_user
   end
   helper_method :logged_in?
 
+  def signed_in?
```

Comment:
This should probably be called `admin_logged_in?` -- then maybe the other method becomes `user_logged_in?`